src/Drupal/Commands/core/MigrateRunnerCommands.php

```diff
@@ -696,6 +749,11 @@ protected function getMigrationPluginManager(): MigrationPluginManagerInterface
      */
     protected function getSourceIdKeys(MigrateIdMapInterface $idMap): array
     {
+        if (iterator_count($idMap) == 0) {
```

Comment:
Sure, it not strictly needed, since it behaves similar to `count()`, but the reason behind it may be just to make it easier to read (and avoid depending remembering details), so I added the change.